mysql-compatibility: add more details (#3796)
pingcap/docs: mysql-compatibility.md

+ Events
+ User-defined functions
+ `FOREIGN KEY` constraints [#18209](https://github.com/pingcap/tidb/issues/18209)

> + Temporary tables

Reviewer: This is already mentioned below as `CREATE TEMPORARY TABLE syntax #1248`.

Reviewer 2: @pepezzzz This is duplicate with [Line 35](https://github.com/pingcap/docs/pull/3796/files#diff-7028919502ad5a35697a652366fdafadR35). What about removing it?

Reviewer 3: Like Forgien Key feature, CREATE TEMPORARY TABLE syntax is compatitable, but TiDB create a real table insteadly. I suggest that  add the issue link after TEMPORARY TABLE feature, and remove the syntax line, keepace with Forgien Key.

Reviewer: SGTM

Reviewer 2: @pepezzzz done in https://github.com/pingcap/docs/pull/3796/commits/6078be93bebd7b847b1a46cf2e7df4bb31408287. PTAL

+ In TiDB, auto-increment columns are only guaranteed to be unique and incremental on a single TiDB server, but they are *not* guaranteed to be incremental among multiple TiDB servers or allocated sequentially. It is recommended that you do not mix default values and custom values. Otherwise, you might encounter the `Duplicated Error` error message.

> + You can use the `tidb_allow_remove_auto_inc` system variable to enable or disable allowing to remove the `AUTO_RANDOM` column attribute. The syntax of removing the column attribute is `alter table modify` or `alter table change`.

Reviewer 2: ```suggestion
+ You can use the `tidb_allow_remove_auto_inc` system variable to allow or forbid removing the `AUTO_INCREMENT` column attribute. The syntax of removing the column attribute is `alter table modify` or `alter table change`.
```

> + You can use the `tidb_allow_remove_auto_inc` system variable to enable or disable the `AUTO_INCREMENT` column attribute that allows removing columns. The syntax of removing the column attribute is `alter table modify` or `alter table change`.

Reviewer 2: @pepezzzz Is this what `通过 `tidb_allow_remove_auto_inc` 系统变量开启或者关闭允许移除列的 `AUTO_INCREMENT` 属性` mean?
```suggestion
+ You can use the `tidb_allow_remove_auto_inc` system variable to enable or disable allowing to remove the `AUTO_RANDOM` column attribute. The syntax of removing the column attribute is `alter table modify` or `alter table change`.
```

Reviewer 3: 启用 tidb_allow_remove_auto_inc 后，才能允许做自增属性移除操作，它是一个保护开关，因为这个操作是不可逆的。非专业意见供参考。
You can use the `tidb_allow_remove_auto_inc` system variable to enable or disable the capability to remove `AUTO_INCREMENT` column attribute.

> + TiDB does not support adding the `AUTO_INCREMENT` column attribute, and it cannot be restored after removing this attribute.

Reviewer 2: ```suggestion
+ TiDB does not support adding the `AUTO_INCREMENT` column attribute, and this attribute cannot be recovered once it is removed.
```

* Multiple operations cannot be completed in a single `ALTER TABLE` statement. For example, it is not possible to add multiple columns or indexes in a single statement. Otherwise, the `Unsupported multi schema change` error might be output.
* Different types of indexes (`HASH|BTREE|RTREE|FULLTEXT`) are not supported, and will be parsed and ignored when specified.
* Adding/Dropping the primary key is unsupported unless [`alter-primary-key`](/tidb-configuration-file.md#alter-primary-key) is enabled.

> * Changing the field type to its superset is unsupported. For example, it does not support changing from `INTEGER` to `VARCHAR`, or from `TIMESTAMP` to `DATETIME`. Otherwise, the error information `Unsupported modify column: type %d not match origin %d` might be output.

Reviewer 2: ```suggestion
* Changing the field type to its superset is unsupported. For example, TiDB does not support changing the field type from `INTEGER` to `VARCHAR`, or from `TIMESTAMP` to `DATETIME`. Otherwise, the error information `Unsupported modify column: type %d not match origin %d` might be output.
```

These differences are documented further in [`ANALYZE TABLE`](/sql-statements/sql-statement-analyze-table.md).

### Limitations of `SELECT` syntax

> - `SELECT ... INTO @variable` syntax is not supported.

Reviewer 2: ```suggestion
- The `SELECT ... INTO @variable` syntax is not supported.
```

> - `SELECT ... GROUP BY ... WITH ROLLUP` syntax is not supported.

Reviewer 2: ```suggestion
- The `SELECT ... GROUP BY ... WITH ROLLUP` syntax is not supported.
```
